### PR TITLE
[docs] document persistence.

### PR DESCRIPTION
**Pull request by chiscoder**
Hello, I added some basic example and documentation about persistence because there is none right now and you have to deep into the code to understand how to do it.

### CODE REVIEW

File: docs/docs/guides/persistence.mdx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+---
+title: State Persistence
+sidebar_label: State Persistence
+---
+
+
+Recoils makes it easy to persist your state.
```
Comment (atanasster) [inline docs/docs/guides/persistence.mdx:7]: Recoil

File: docs/docs/guides/persistence.mdx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+```
+
+*Storage* could be [*LocalStorage*](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage), *[AsyncStorage](https://github.com/react-native-community/react-native-async-storage)* or whichever storage you like.
+
+* *atomValues*: Gives you information about the current state of your atoms.
+* *modifiedAtoms* Gives you a map containing the modified atoms.
+* *transactionMetaData* Gives you metadata about the current change.
+
+
+After you ensure that you're saving yout state to your storage, you need to recover it, it's easy using the prop *initializeState* on *RecoilRoot* component.
```
Comment (atanasster) [inline docs/docs/guides/persistence.mdx:30]: ‘saving your state’

File: docs/docs/guides/persistence.mdx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+It's a function that give you set method that allows you to update atoms using their key.
```
Comment (atanasster) [inline docs/docs/guides/persistence.mdx:31]: ‘a function that provides a set method to update atoms given a key’

**Comment by cgomezmendez** [comment]
Updated the PR with the fixes

**Comment by drarmstr** [comment]
Thank you so much @cgomezmendez for taking the initiative to dive into the code and figuring out how to do persistence.  I only hesitate to publish docs on this API right now since I'm actively working on cleaning up the API and would kinda like to avoid whiplash so soon; hence them being marked as _UNSTABLE.  But, maybe it's ok to get this up with some stronger disclaimers that this is how people can explore right now, though expect it to change in the near future.

**Comment by cgomezmendez** [comment]
> Thank you so much @cgomezmendez for taking the initiative to dive into the code and figuring out how to do persistence. I only hesitate to publish docs on this API right now since I'm actively working on cleaning up the API and would kinda like to avoid whiplash so soon; hence them being marked as _UNSTABLE. But, maybe it's ok to get this up with some stronger disclaimers that this is how people can explore right now, though expect it to change in the near future.

What about this at the beginning

**Note that the useTransactionObservation API is NOT stable, so use at your own risk. They may change at any time (as we're experimenting with them)**

File: docs/docs/guides/persistence.mdx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            const promises = keys.map((key) =>
+                Storage.getItem(key)
+            );
+            Promise.all(promises).then((values) => {
+                for (let i = 0; i < promises.length; i++) {
+                    const key = keys[i];
+                    const value = JSON.parse(values[i]).value;
+                    set({ key }, value);
+                }
+                setIsReady(true);
```
Comment (alexandrzavalii) [inline docs/docs/guides/persistence.mdx:48]: what `setIsReady` is doing? should there be a useState?

Comment (chiscoder) [reply docs/docs/guides/persistence.mdx:48]: Yes, it's a useState setter, should I remove it or add the useState before the method?

Comment (alexandrzavalii) [reply docs/docs/guides/persistence.mdx:48]: I think its better to remove it,

Comment (chiscoder) [reply docs/docs/guides/persistence.mdx]: Done.

**Pull request merged**